extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1147,17 +1155,50 @@ export class AmpA4A extends AMP.BaseElement {
    * @param {!Window} iframeWin
    */
   registerExpandUrlParams_(iframeWin) {
-    iframeWin.document.documentElement.addEventListener('click',
-        this.maybeExpandUrlParams_.bind(this), /* capture */ true);
+    // Mapping of message origin to parameters allowing for replacement.
+    // Multiple messages from same origin result in last value being
+    // overwritten.
+    /** @type {Object<string, Object<string, string>>} */
+    const messages = {};
+    // Listen for messages to creative friendly frame.
+    this.unlisteners_.push(listen(iframeWin, 'message', evt => {
+      if (!evt || !evt.source || !evt.origin) {
+        return;
+      }
+      // Only allow messages from child iframe whose parent is an
+      // amp-signal-collection-frame element.
+      const frames = iframeWin.document
```

Comment:
The reason not to cache is that attachment of the iframe within amp-signal-collection-frame is done within layoutCallback of that element.  Given it will have a lower priority (2) it is likely it will be appended AFTER registerExpandUrlParams_ executes.  I also want to allow for the possibility that multiple amp-signal-collection-frame elements exist within the AMP creative so its not sufficient to use the result of querySelectorAll after the first post message.  I suppose I could verify the count of iframes = count of total amp signal collection frames and cache at that time... just not sure its worth it.  Thoughts?